### PR TITLE
Add Test Projects for DCS/DCJS's ReflectionOnly mode.

### LINKED ISSUE #10967: Create Test Projects for Testing Reflection Based DCS/DCJS

### PR DESCRIPTION
**Pull request by shmao**
Add test projects for testing DCS/DCJS's reflection based serialization.

By default, the serialization mode is set to `ReflectionAsBackup`. The change uses reflection to set the mode to `ReflectionOnly`.

Fix #10967.

/cc: @mconnew @mlacouture @huanwu @zhenlan

### CODE REVIEW

**Comment by shmao** [comment]
Can you please have a look at the PR? The fix is to add test coverage for reflection based serialization. Thanks!

/cc: @mconnew @mlacouture @huanwu @zhenlan

File: src/System.Runtime.Serialization.Json/tests/ReflectionOnly/System.Runtime.Serialization.Json.ReflectionOnly.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <DefineConstants>$(DefineConstants);ReflectionOnly</DefineConstants>
+    <RootNamespace>System.Runtime.Serialization.Json.ReflectionOnly.Tests</RootNamespace>
+    <AssemblyName>System.Runtime.Serialization.ReflectionOnly.Json.Tests</AssemblyName>
```
Comment (zhenlan) [inline src/System.Runtime.Serialization.Json/tests/ReflectionOnly/System.Runtime.Serialization.Json.ReflectionOnly.Tests.csproj]: Do you like to keep the AssemblyName consistent with RootNamespace or is this intentional?

File: src/System.Runtime.Serialization.Xml/tests/System.Runtime.Serialization.Xml.Tests.settings.targets [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <MinimumVisualStudioVersion>11.0</MinimumVisualStudioVersion>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>System.Runtime.Serialization.Xml.Tests</RootNamespace>
+    <AssemblyName>System.Runtime.Serialization.Xml.Tests</AssemblyName>
+    <ProjectGuid>{30CAB353-089E-4294-B23B-F2DD1D945654}</ProjectGuid>
```
Comment (zhenlan) [inline src/System.Runtime.Serialization.Xml/tests/System.Runtime.Serialization.Xml.Tests.settings.targets]: I think you don't want to keep RootNamespace, AssemblyName and ProjectGuid here?

Comment (zhenlan) [inline src/System.Runtime.Serialization.Xml/tests/System.Runtime.Serialization.Xml.Tests.settings.targets]: I think you don't want to keep RootNamespace, AssemblyName and ProjectGuid here?

File: src/System.Runtime.Serialization.Json/tests/ReflectionOnly/System.Runtime.Serialization.Json.ReflectionOnly.Tests.csproj (hunk 1 above)
Comment (zhenlan) [inline src/System.Runtime.Serialization.Json/tests/ReflectionOnly/System.Runtime.Serialization.Json.ReflectionOnly.Tests.csproj]: Do you like to keep the AssemblyName consistent with RootNamespace or is this intentional?

Comment (shmao) [inline src/System.Runtime.Serialization.Json/tests/ReflectionOnly/System.Runtime.Serialization.Json.ReflectionOnly.Tests.csproj]: Changed the RootNamespace. Thanks!

File: src/System.Runtime.Serialization.Xml/tests/System.Runtime.Serialization.Xml.Tests.settings.targets (hunk 2 above)
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/System.Runtime.Serialization.Xml.Tests.settings.targets]: You are right. Removed them.

**Comment by shmao** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test 

One of the System.Collections tests failed, 
>  System.Collections.Tests.CaseInsensitiveHashCodeProviderTests.Default_GetHashCodeCompare(a: "HELLO", b: "HELLO", expected: True)

**Comment by shmao** [comment]
Thanks for reviewing the changes, @mlacouture and @zhenlan . Merged the PR.

**Pull request merged**